warehouse/views.py

```diff
@@ -192,7 +193,11 @@ def search(request):
     if request.params.getall("c"):
         query = query.filter("terms", classifiers=request.params.getall("c"))
 
-    page_num = int(request.params.get("page", 1))
+    try:
+        page_num = int(request.params.get("page", 1))
+    except ValueError:
+        raise HTTPBadRequest
```

Comment:
It'd probably be nice to add a message to this, like ``raise HTTPBadRequest("'page' must be an integer, was {!r}".format(request.params["page"])`` or something like that. I'm pretty sure Pyramid will expose that information in the response so people can see what they did wrong.